scripts/example_script.py

```diff
@@ -0,0 +1,238 @@
+
+    # create a folder under this collection or
+    # load a preexisting folder that exists with the same name and parent
+
+    newFolderData = gc.load_or_create_folder(new_folder_name,
+                                             collectionId, 'collection')
+    newFolderId = newFolderData['_id']
+
+    # get a list of all items within each folder of the collection
+    itemList = recurse_get_resource(gc, collectionId, 'item', 'collection')
+
+    tempImageId = None
+    print('Listing all %s items in the collection %s' % (img_type,
+                                                         collectionName))
+    # create a list to store id of all svs items
+    img_list = []
+    for item in itemList:
+        if img_type in item['name']:
+            tempImageId = item['_id']
+            print('image name %s ' % (item['name']))
+            img_list.append(tempImageId)
+            make_large_image(gc, tempImageId)
+
+    # now create PATIENT folders inside our new folder
+
+    patientFolderList = []
+    patient_folder_name = 'Patient%d'
+
+    for i in range(1, 4):
+        tempName = patient_folder_name % (i)
+
+        patientFolderList.append(
+                gc.load_or_create_folder(tempName, newFolderId, 'folder'))
+
+    # now let us move an image into the PATIENT1 folder
+
+    patient1Folder = patientFolderList[0]
+    patient1FolderId = patient1Folder['_id']
+
+    srcImageId = img_list[0]
+
+    moveParams = {
+        'resources': json.dumps({
+            'item': [str(srcImageId)]}),
+        'parentType': 'folder',
+        'parentId': patient1FolderId
+    }
+
+    try:
+        gc.put('resource/move', moveParams)
+    except HttpError as err:
+        print_http_error(err)
+        print ('exited')
+        exit()
+
+    # now let us edit the PATIENT1 folder
+
+    patient1MetaData = {'PatientAge': 22, 'PatientDeceased': True}
+
+    try:
+        gc.addMetadataToFolder(patient1FolderId, patient1MetaData)
+    except HttpError as err:
+        print_http_error(err)
+        print ('exited')
+        exit()
+
+    # let us add meta data to the image we moved
+    # note how the id remains the same
+    newSrcImageMetaData = {'SlideSource': 'FromTCGA', 'SlideType': 'DX'}
+    try:
+        gc.addMetadataToItem(srcImageId, newSrcImageMetaData)
+    except HttpError as err:
+        print_http_error(err)
+        print ('exited')
+        exit()
+
+    # find all folders determine if whether each folder is empty or not
+    allFolders = []
+
+    allFolders = recurse_get_resource(gc, collectionId,
+                                      'folder', 'collection')
+
+    # if the folder size is zero then delete it
+    for folder in allFolders:
+
+        if folder['size'] == 0 and isEmpty(gc, folder['_id']):
+            try:
+                print('will delete folder %s' % folder['name'])
+
+                gc.delete('folder/%s' % folder['_id'])
+
+            except HttpError as err:
+                print_http_error(err)
```

Comment:
Shouldn't need the `try` block here, since `main` is wrapped in one. I don't think it's that important to have the extra `print` statement on an error here.